### ISSUE TITLE
Incorrect detection of wrong URLs at notification time

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#1652]
It seems that Orion (as in 0.26.1) allows create subscriptions which reference is port 0 (ilegal, as port ranges from 1 to 65535):

```
time=2016-01-11T12:29:20.204CET | lvl=INFO | trans=1452511738-622-00000000337 | function=LM_TRANSACTION_START | comp=Orion | msg=logMsg.h[1792]: Starting transaction to foo.bar.net:0/ngsi10/notifyContext
time=2016-01-11T12:29:20.204CET | lvl=ERROR | trans=1452511738-622-00000000337 | function=httpRequestSendWithCurl | comp=Orion | msg=httpRequestSend.cpp[220]: Runtime Error (port is ZERO)
time=2016-01-11T12:29:20.204CET | lvl=INFO | trans=1452511738-622-00000000337 | function=LM_TRANSACTION_END | comp=Orion | msg=logMsg.h[1815]: Transaction ended
```

Thus, probably reference should be checked at subscription creation/update time, in order to detect ilegal port (and maybe other URL checkings).

**Comment by fgalan** [comment]
Taking into account the content of the DB:

```
"reference" : "http://fiwarenr.eu-gb.mybluemix.net/ngsi10/notifyContext/dev-fixed",
```

The proble is not related with ilegal port at subscription time, but with incorrectly setting the default to 80.

**Comment by fgalan** [comment]
After more research, the problem is due the following strings are not detected as wrong URL:

```
http://http://XXX:1234/path
http://XXX:/path
```

Adjusting parseUrl() solves the issue. I'll take care.

**Comment by fgalan** [comment]
Fixed in PR #1653
